### DISCUSSION TITLE
Question: Is PersistenceId necessary in WinAppSdk/WinUI3?

### DISCUSSION THREAD

**Discussion by 0x5bfa** [dotMorten/WinUIEx#179] [Q&A] [answered]
I'm experimenting with an app that uses WinUIEx.
Almost of all are already implemented within WinAppSdk expect for min width, min height and persistence id. While min width and height can be set using WndProc when window tries to get the info with message code WM_GETMINMAXINFO, I couldn't find any alternatives for persistence id.

I looked into the WinUIEx source code and I find actual store/restore routine here:

https://github.com/dotMorten/WinUIEx/blob/ea41e17bc3ecba48491db9920a9ed7348f62d44b/src/WinUIEx/WindowManager.cs#L468

If I understand this correctly, this stores WINDOWPLACEMENT struct data in case dev customizes that data in their code using SetWindowPlacement.
If I don't set that, is it necessary to set?

**Comment by dotMorten** [comment, marked as answer]
It’s not necessary. You only need to set it if you want the app to launch where it was closed.

**Reply by 0x5bfa** [reply]
Thanks!
